contracts/impls/trustable/WitnetRequestBoardTrustableBase.sol

```diff
@@ -0,0 +1,497 @@
+    // --- Overrides 'Upgradable' -------------------------------------------------------------------------------------
+
+    /// Initialize storage-context when invoked as delegatecall. 
+    /// Should fail when trying to initialize same instance more than once.
+    function initialize(bytes memory _initData) virtual external override {
+        address _owner = _state().owner;
+        if (_owner == address(0)) {
+            // set owner if none set yet
+            _owner = msg.sender;
+            _state().owner = _owner;
+        } else {
+            // only owner can initialize:
+            require(msg.sender == _owner, "WitnetRequestBoardTrustableBase: only owner");
+        }        
+
+        if (_state().base != address(0)) {
+            // current implementation cannot be initialized more than once:
+            require(_state().base != base(), "WitnetRequestBoardTrustableBase: already initialized");
+        }        
+        _state().base = base();
+
+        emit Initialized(msg.sender, base(), codehash(), version());
+
+        // Do actual base initialization:
+        setReporters(abi.decode(_initData, (address[])));
+    }
+
+    /// Tells whether provided address could eventually upgrade the contract.
+    function isUpgradableFrom(address _from) external view override returns (bool) {
+        address _owner = _state().owner;
+        return (
+            // false if the WRB is intrinsically not upgradable
+            isUpgradable() && (                
+                _owner == address(0) ||
+                _owner == _from
+            )
+        );
+    }
+
+
+    // ================================================================================================================
+    // --- Full implementation of 'IWitnetRequestBoardAdmin' -----------------------------------------------------------------
+
+    /// Gets admin/owner address.
+    function owner()
+        public view
+        override
+        returns (address)
+    {
+        return _state().owner;
+    }
+
+    /// Transfers ownership.
+    function transferOwnership(address _newOwner)
+        external
+        virtual override
+        onlyOwner
+    {
+        address _owner = _state().owner;
+        if (_newOwner != _owner) {
+            _state().owner = _newOwner;
+            emit OwnershipTransferred(_owner, _newOwner);
+        }
+    }
+
+
+    // ================================================================================================================
+    // --- Full implementation of 'IWitnetRequestBoardAdminACLs' -------------------------------------------------------------
+
+    /// Tells whether given address is included in the active reporters control list.
+    /// @param _reporter The address to be checked.
+    function isReporter(address _reporter) public view override returns (bool) {
+        return _acls().isReporter_[_reporter];
+    }
```

Comment:
This whole reporters feature is over-engineered and deviates from the requirements, as we have never stated a need for more than one reporter. But hey, it works, so I have no problem with it for now.

Reply:
Ok, sorry, it had a short and obvious implementation so I couldn't help it.

Future implementation, or this one, could eventually deprecate these kind of admin methods with no back-compatibility issues with user contracts, as these admin interfaces are in fact not inherited by the abstract `WitnetRequestBoard`.